README.md

```diff
@@ -173,3 +173,124 @@ to determine that should this request be counted into rate-limit
 or not based on information given by `Request` object
 (i.e., Skip counting rate-limit on some route) and the key of the given request,
 by default, this will always return `false` which means counted everything.
+
+### getServer
+
+`(app: Elysia) => MaybePromise<Elysia | NeedRequestIP>`
+
+Default: `(app) => app`
+
+```ts
+import { Elysia } from 'elysia'
+import { rateLimit } from 'elysia-rate-limit'
+import bearer from '@elysiajs/bearer'
+
+class appInstance {
+  static server: Server | null
+}
+
+const setup = new Elysia({
+  name: 'setup',
+}).use(bearer())
+
+const userList = new Elysia({
+  name: 'userListRoute',
+})
+  .use(setup)
+  .use(
+    rateLimit({
+      getServer: _ => appInstance,
+      scoping: 'scoped',
+      max: 10,
+      duration: 100,
+    })
+  )
+  .get('/user', ({ bearer }) => {
+    return user.list()
+  })
+
+const userAdd = new Elysia({
+  name: 'userAddRoute',
+})
+  .use(setup)
+  .use(
+    rateLimit({
+      getServer: _ => appInstance,
+      scoping: 'scoped',
+      max: 2,
+      duration: 1000,
+    })
+  )
+  .post('/user', ({ body }) => {
+    return user.add(body)
+  })
+
+const app = new Elysia({
+  name: 'app',
+})
+  .use(helmet())
+
+  .use(userAdd)
+  .use(userList)
+
+  .listen(
+    {
+      port: 8081,
+    },
+    ({ development, hostname, port }) => {
+      console.log(
+        `🦊 Elysia is running at ${hostname}:${port} ${
+          development ? '🚧 in development mode!🚧' : ''
+        }`
+      )
+    }
+  )
+
+appInstance.server = app.server
+```
+
+### throwOnError
+
+`Error`
+
+Throw this object when rate-limit reached
+
+Read more :
+
+- [Error Provider - ElysiaJS | ElysiaJS](https://elysiajs.com/validation/error-provider.html#onerror)
+- [Error Handling - ElysiaJS | ElysiaJS](https://elysiajs.com/life-cycle/on-error.html)
+
+```ts
+import { Elysia } from 'elysia'
+import { rateLimit } from 'elysia-rate-limit'
+import { HttpStatusEnum } from 'elysia-http-status-code/status'
+
+export class rateLimitError extends Error {
+  constructor(
+    public message: string = 'TOO_MANY_REQUESTS',
+    public detail: string = '',
+    public status: number = HttpStatusEnum.HTTP_429_TOO_MANY_REQUESTS // or just 429
+  ) {
+    super(message)
+  }
+}
+
+new Elysia()
+  .use(
+    rateLimit({
+      throwOnError: new rateLimitError(),
+    })
+  )
+  // use with error hanlder
+  .error({
+    RateLimit: rateLimitError,
+  })
+  .onError({ as: 'global' }, ({ code }) => {
+    switch (code) {
+      case 'RateLimit':
+        return code
+        break
+    }
+  })
+  .listen(3000)
+```
```

Comment:
I would refactor `responseMessage`, and `responseCode` into something else. removing both and make an option `errorResponse: string | Response` instead.

so if `typeof errorResponse` is a string, then it return a stringified message with status 429 as a standard, otherwise if you want anything more than this return a web standard `Response` object.

this way is much better